### PR TITLE
 Support T-SQL sp_linkedservers stored procedure

### PR DESCRIPTION
**Pull request by basasairohan**
### Description

This commit adds support for sp_linkedservers stored procedure.
sp_linkedservers has been implemented such that internally it uses sys.servers view.

Signed-off-by: Sai Rohan Basa [bsrohan@amazon.com](mailto:bsrohan@amazon.com)

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** N/A

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** Yes

* **Client tests -** Yes

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 1]
```diff
@@ -2891,3 +2891,20 @@ END;
...
+    	product AS SRV_PRODUCT, 
+    	data_source AS SRV_DATASOURCE,
+		provider_string AS SRV_PROVIDERSTRING,
+		location AS SRV_LOCATION,
+		catalog AS SRV_CAT 
+	FROM sys.servers
+	ORDER BY SRV_NAME
+END;
+$$ LANGUAGE 'pltsql';
+GRANT EXECUTE ON PROCEDURE sys.sp_linkedservers TO PUBLIC;
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: no newline

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 2]
```diff
@@ -2891,3 +2891,20 @@ END;
...
+
+CREATE OR REPLACE PROCEDURE sys.sp_linkedservers()
+AS $$
+BEGIN
+    SELECT 
+    	name AS SRV_NAME, 
+    	provider AS SRV_PROVIDERNAME, 
+    	product AS SRV_PRODUCT, 
+    	data_source AS SRV_DATASOURCE,
+		provider_string AS SRV_PROVIDERSTRING,
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: fix indentation

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql [hunk 3]
```diff
@@ -1106,6 +1106,23 @@ RETURNS INTEGER AS
...
 
+CREATE OR REPLACE PROCEDURE sys.sp_linkedservers()
+AS $$
+BEGIN
+    SELECT 
+    	name AS SRV_NAME, 
+    	provider AS SRV_PROVIDERNAME, 
+    	product AS SRV_PRODUCT, 
+    	data_source AS SRV_DATASOURCE,
+		provider_string AS SRV_PROVIDERSTRING,
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql]: fix indentation

File: test/JDBC/input/linked_servers-vu-verify.sql [hunk 4]
```diff
@@ -13,3 +13,6 @@ GO
 
 SELECT * FROM sys_linked_servers_vu_prepare__sys_linked_logins_view
 GO
+
+EXEC sp_linkedservers
+GO
```
Comment (thephantomthief) [inline test/JDBC/input/linked_servers-vu-verify.sql]: add a new line at EOF

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 5]
```diff
@@ -2891,3 +2891,20 @@ END;
 $$
 LANGUAGE 'pltsql';
 GRANT ALL ON PROCEDURE sys.sp_sproc_columns_100 TO PUBLIC;
+
+CREATE OR REPLACE PROCEDURE sys.sp_linkedservers()
+AS $$
+BEGIN
+    SELECT 
+    	name AS SRV_NAME,
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: I think case would not be persisted. You will have to enclose the column names in quotes

**Review by thephantomthief — DISMISSED** [review]

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 6]
```diff
@@ -2903,3 +2903,21 @@ END;
 $$
 LANGUAGE 'pltsql';
 GRANT ALL ON PROCEDURE sys.sp_sproc_columns_100 TO PUBLIC;
+
+CREATE OR REPLACE PROCEDURE sys.sp_linkedservers()
+AS $$
+BEGIN
+    SELECT 
+		name AS "SRV_NAME", 
+		provider AS "SRV_PROVIDERNAME",
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql]: sys.servers. provider is of sys.sysname but we need to return nvarchar(128). Same is true for SRV_PRODUCT

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**